final_task/rss_reader/collect_news.py

```diff
@@ -0,0 +1,51 @@
+import feedparser
+import html
+
+from . import logg, converter, news_parser
+
+
+def collectNews(limit, tojson, tohtml, topdf, color, source):
+    '''
+    1. cache news
+    2. create html or pdf document
+    3. or return news in json or normal format
+    '''
+    news = list()
+
+    channel = feedparser.parse(source)
+    news.append(color[0] + "Feed: " + channel.feed.title + '\n')
+    limit = limit or len(channel.entries)
+
+    news_parser.cacheNews(channel)
+
+    if (tohtml or topdf):
+        html_doc = converter.createHtmlStructure(channel, limit, tohtml, topdf)
+        return html_doc
+    else:
+        for index, item in enumerate(channel.entries):
+            if (index == limit):
+                break
```

Comment:
It is better to use slices instead of comparing index with limit on each iteration